File: 1.1/src/c/cz/mg/vulkan/PFN_vkQueueBindSparse.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkQueueBindSparse_callNative(JNIEnv* env, jclass jc, jlong address, jlong queue, jint bindInfoCount, jlong pBindInfo, jlong fence){
    (void)env;
    (void)jc;
    PFN_vkQueueBindSparse f = (PFN_vkQueueBindSparse)jniLongToFunctionPointer(address);
    return f(
        *((VkQueue*)jniLongToPointer(queue)),
        (uint32_t)bindInfoCount,
        ((VkBindSparseInfo*)jniLongToPointer(pBindInfo)),
        *((VkFence*)jniLongToPointer(fence))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkQueueBindSparse.java
package cz.mg.vulkan;

public class PFNvkQueueBindSparse extends VkFunctionPointer {
    public PFNvkQueueBindSparse() {
    }

    protected PFNvkQueueBindSparse(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkQueueBindSparse(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkQueueBindSparse(long value) {
        setValue(value);
    }

    public PFNvkQueueBindSparse(VkInstance instance) {
        super(instance, new VkString("vkQueueBindSparse"));
    }

    public void call(VkQueue queue, VkUInt32 bindInfoCount, VkBindSparseInfo pBindInfo, VkFence fence, VkResult rval){
        callNative(getValue(), queue != null ? queue.getVkAddress() : VkPointer.getNullAddressNative(), bindInfoCount != null ? bindInfoCount.getVkAddress() : VkPointer.getNullAddressNative(), pBindInfo != null ? pBindInfo.getVkAddress() : VkPointer.NULL, fence != null ? fence.getVkAddress() : VkPointer.getNullAddressNative(), rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long queue, long bindInfoCount, long pBindInfo, long fence, long rval);


    public int call(VkQueue queue, int bindInfoCount, VkBindSparseInfo pBindInfo, VkFence fence){
        return callSimplifiedNative(getValue(), queue != null ? queue.getVkAddress() : VkPointer.getNullAddressNative(), bindInfoCount, pBindInfo != null ? pBindInfo.getVkAddress() : VkPointer.NULL, fence != null ? fence.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native int callSimplifiedNative(long vkaddress, long queue, int bindInfoCount, long pBindInfo, long fence);

}
